runtime/major_gc.c

```diff
@@ -946,6 +946,10 @@ static void mark_slice_darken(struct mark_stack* stk, value child,
   /* This part of the code is duplicated in do_some_marking for performance
    * reasons.
    * Changes here should probably be reflected in do_some_marking. */
+  /* Annotating a acquire barrier on `p` because TSAN does not realise
+   * a happens-before relatioship established due to address dependencies
+   * with the read in shared_heap.c allocation (#12894) */
+    CAML_TSAN_ANNOTATE_HAPPENS_AFTER(Hp_val(child));
```

Comment:
```suggestion
   * Changes here should probably be reflected in do_some_marking. */
   
  /* Annotating an acquire barrier on the header because TSan does not see the
   * happens-before relationship established by address dependencies with
   * initializing writes in shared_heap.c allocation (#12894) */
    CAML_TSAN_ANNOTATE_HAPPENS_AFTER(Hp_val(child));
```
The comment refers to `p` which is probably an unintentional leftover from copy-pasting.

I suggest to add a blank line to separate from the previous unrelated comment. The above also suggests a few minor word tweaks that make the sentence clearer IMO. (And it’s not a read in shared_heap.c, it’s two writes)